src/payment/strategies/external/external-payment-strategy.ts

```diff
@@ -53,7 +53,7 @@ export default class ExternalPaymentStrategy implements PaymentStrategy {
         return Promise.resolve(this._store.getState());
     }
 
-    private _isAdditionalActionRequired(error: RequestError): boolean {
+    protected _isAdditionalActionRequired(error: RequestError): boolean {
```

Comment:
It's a bit odd that this method is named `_isAdditionalActionRequired` when it only detects offside redirects (a very specific additional action).

It also validates the shape of the request error, but then just returns a boolean. Meaning the code which calls it misses out on any type information (e.g. all the `Object is of type 'unknown'.` errors in `HummPaymentStrategy`)

This might be better changed into a type guard that turns the response from an `unknown` to a `OffsiteRedirectResponse` entity.

Something like:

```ts
    protected _isOffsiteRedirectResponse(response: unknown): response is OffsiteRedirectResponse {
        if (typeof response !== 'object' || response === null) {
            return false;
        }

        const partialResponse: Partial<OffsiteRedirectResponse> = response;

        if (!partialResponse.body) {
            return false;
        }

        const partialBody: Partial<OffsiteRedirectBody> = partialResponse.body;

        return partialBody.status === 'additional_action_required'
            && !!partialBody.additional_action_required
            && partialBody.additional_action_required.type === 'offsite_redirect';
    }
```

That way code which uses it benefits from the discerned type information (no more TypeScript errors) and we can write our code a bit more intuitively:

```ts
 } catch (error) {
            if (this._isOffsiteRedirectResponse(error)) {
                // do things safely knowing error is a OffsiteRedirectResponse
                // doing redirecting etc
            } esle {
                // handle it being an unknown error
            }
        }
```